File: 11.Stack/Javascript/1.PushPopTop.js
function stackOperations(operations, n) {
    const stack = [];
    const result = [];

    for (let operation of operations) {
        if (operation.startsWith("1")) {
            const [, x] = operation.split(" ");
            stack.push(parseInt(x));
        } else if (operation === "2") {
            if (stack.length > 0) {
                stack.pop();
            }
        } else if (operation === "3") {
            if (stack.length > 0) {
                result.push(stack[stack.length - 1].toString());
            } else {
                result.push("Empty!");
            }
        }
    }

    return result.join("\n");
}

const operations = ["1 15", "1 20", "2", "3", "2", "3"];
const n = 5;

const res = stackOperations(operations, n);
console.log(res);

// Output:
// 15
// Empty!

File: 11.Stack/Python/1.PushPopTop.py
def stack_operations(operations,n):

    stack = []
    result = []

    for operation in operations:
        if operation[0] == "1":
            _,x = operation.split()
            stack.append(int(x))
        elif operation[0] == "2":
            if stack:
                stack.pop() 
        elif operation[0] == "3":
            if stack:
                result.append(str(stack[-1]))
            else:
                result.append("Empty!")

    return "\n".join(result)

operations = ["1 15", "1 20", "2", "3", "2","3"]
n = 5

res = stack_operations(operations,n)
print(res)

# Output:-
# 15    
# Empty!